### PR TITLE
Added the feature : Added the clickable titles which collapse and expand on side bar nav

### LINKED ISSUE #53: [Feat] Expand and collapse batches
### Description

By Default, All batches are expanded. It would be great if user can expand and collapse batch (like if you want to learn HTML, CSS, JS, you don't need to see other batches like Hire or Build)

image

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Akash-pugazh**
## Fixes Issue : Feat added : Added the clickable titles on Side bar nav 

Closes #53 

## Changes proposed

Made changes on the sidebar-nav.tsx file : Making the UI element on sidebar as accordion

Edit : I've noticed a older issue on making the title clickable to open the content and Added in this second commit Please feel free to go through the code

Edit 2 : Sorry Ive Re-Submitting this due to merge conflicts. 😊

## Quick note : 

I have also made a little change on reducing the padding bottom from 8 to 4 (Line: 25), because it really looks more spaced when its large, Feel free to modify or I'll make sure to edit and PR again. 😊

##### Some little changes on UI code

Main Issue Fix :

Screenshot (14)

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [x] My change requires changes to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

Screenshot (15)

Screenshot (16)

## Note to reviewers

This is minor change and also my first contribution, Thank you for this opportunity, will contribute more from here 💕

### CODE REVIEW

**Comment by Vishal-raj-1** [comment]
In dark theme, title color should be white

Fix this

image

File: components/layout/sidebar-nav.tsx [hunk 1]
```diff
@@ -22,36 +22,49 @@ export function DocsSidebarNav({ items }: DocsSidebarNavProps) {
...
-                </>
-              ))}
-            </div>
-          ) : null}
+        <div key={index} className={cn("pb-4")}>
+          <Accordion type="single" collapsible>
+            <AccordionItem className="border-b-0" value="item-title">
+              <div>
+                <AccordionTrigger className="py-2 hover:no-underline">
+                  {item.title}
```
Comment (Vishal-raj-1) [inline components/layout/sidebar-nav.tsx:30]: title style is missing, "mb-1 rounded-md py-1 text-base font-medium dark:text-white"

**Review by Vishal-raj-1 — APPROVED** [review]

**Review by Vishal-raj-1 — CHANGES_REQUESTED** [review]
fix dark theme style

File: components/layout/sidebar-nav.tsx (hunk 1 above)
Comment (Akash-pugazh) [reply components/layout/sidebar-nav.tsx:30]: I'll modify 😊👍

**Comment by Akash-pugazh** [comment]
Hello there Maintainers, I've made respective requested changes, Thankyou for guiding me out😊

Some Screenshots:  

Screenshot (19)

Please report if any other issues

**Review by Vishal-raj-1 — APPROVED** [review]

**Pull request merged**